plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/plan/CodeModernizerPlanEditor.kt

```diff
@@ -3,178 +3,495 @@
+                addHyperlinkListener { he ->
+                    if (he.eventType == HyperlinkEvent.EventType.ACTIVATED) {
+                        SwingUtilities.invokeLater { rootPanel.verticalScrollBar.setValue(0) }
+                    }
+                }
+                isEditable = false
+                isOpaque = false
+                alignmentX = Component.RIGHT_ALIGNMENT
+            }
+        scrollPanel.add(scrollPane, BorderLayout.CENTER)
+        scrollPanel.add(scrollIcon, BorderLayout.EAST)
+        return scrollPanel
     }
 
-    private fun transformationPlanPanel(plan: TransformationPlan) = JPanel(GridBagLayout()).apply {
-        val stepsIntroTitle = JLabel(message("codemodernizer.migration_plan.body.steps_intro_title")).apply {
-            font = font.deriveFont(
-                CodeModernizerUIConstants.FONT_CONSTRAINTS.BOLD,
-                CodeModernizerUIConstants.PLAN_CONSTRAINTS.TRANSFORMATION_STEP_TITLE_FONT_SIZE
-            )
-            border = CodeModernizerUIConstants.STEP_INTRO_TITLE_BORDER
+    private fun getFormattedString(actionString: String) =
+        when (actionString) {
+            "linesOfCode" -> message("codemodernizer.migration_plan.body.info.lines_of_code_message")
+            "plannedDependencyChanges" -> message("codemodernizer.migration_plan.body.info.dependency_replace_message")
+            "plannedDeprecatedApiChanges" -> message("codemodernizer.migration_plan.body.info.deprecated_code_message")
+            "plannedFileChanges" -> message("codemodernizer.migration_plan.body.info.files_changed_message")
+            "dependencyName" -> message("codemodernizer.migration_plan.body.info.dependency_name_column")
+            "action" -> message("codemodernizer.migration_plan.body.info.action_column")
+            "currentVersion" -> message("codemodernizer.migration_plan.body.info.current_version_column")
+            "targetVersion" -> message("codemodernizer.migration_plan.body.info.target_version_column")
+            "relativePath" -> message("codemodernizer.migration_plan.body.info.file_column")
+            "apiFullyQualifiedName" -> message("codemodernizer.migration_plan.body.info.deprecated_code_column")
+            "numChangedFiles" -> message("codemodernizer.migration_plan.body.info.changed_files_column")
+            else -> actionString
         }
-        val stepsIntro = JPanel().apply {
-            layout = BoxLayout(this, BoxLayout.Y_AXIS)
-            add(stepsIntroTitle, CodeWhispererLayoutConfig.inlineLabelConstraints)
-            border = CodeModernizerUIConstants.STEP_INTRO_BORDER
+
+    // use parsed MD string stored in step 0 progress updates from GetPlan response to create table
+    private fun createTable(stepTable: PlanTable): JTable {
+        val columnsVector = Vector(stepTable.columns)
+        columnsVector.forEachIndexed { index, columnName ->
+            columnsVector[index] = getFormattedString(columnName)
```

Comment:
do we actually need vector?
```
val columnsVector = stepTable.columns.map { getFormattedString(it) }

val data = stepTable.rows.map { row ->
    stepTable.columns.map { col ->
        row.getValueForColumn(col)
    }
}
```

Reply:
Yes because of the `DefaultTableModel` below. Looks like I could also use `.toTypedArray<Any>()` if you think that would be better.